src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -143,20 +143,31 @@ internal static bool TryExtractBaggage(string[] baggageCollection, out Dictionar
                         break;
                     }
 
-                    if (NameValueHeaderValue.TryParse(pair, out NameValueHeaderValue baggageItem))
+                    if (pair.IndexOf('=') < 0)
                     {
-                        if (string.IsNullOrEmpty(baggageItem.Name) || string.IsNullOrEmpty(baggageItem.Value))
-                        {
-                            continue;
-                        }
+                        continue;
+                    }
+
+                    var parts = pair.Split(new[] { '=' }, 2);
```

Comment:
Could we avoid "new array" each time? I wonder if `pair.Split('=', 2)` would just work?

Reply:
You cannot pass the only one char in this particular overload. But it's a good point, thank you. I will put this array to the static field.

Reply:
Also
```
foreach (var pair in item.Split(','))
```
This one also makes an array under the hood since the signature is `Split(params char[] ...)`

Reply:
> static sounds fine to me, curious if there is a better way (I'm not C# expert but gut was telling me there got to be a way 😄)

Yeah, `static` means it will be only one object to the whole application. Otherwise, the field will be created for each `BaggagePropagator` object. :)